ODLA/platforms/dnnl/odla_dnnl.cc

```diff
@@ -644,8 +644,26 @@ odla_value odla_PRelu(odla_value input, odla_value slope,
   //                     = relue(input) - relu(mul(input, mul(-1, slope))
 
   auto relu_v = odla_Relu(input, nullptr);
-  auto neg_slop = unary_eltwise_op(dnnl::algorithm::eltwise_linear, slope, -1.f,
-                                   0.f, nullptr);
+
+  const auto& dims_input = input->shape;
+  const auto& dims_slope = slope->shape;
+  odla_value new_slope = slope;
+
+  // workaround: the slope shape may be different from input shape
+  // and input shape is not nhwc. so we change slope shape to {1, c, 1, 1}
+  if (dims_input.size != dims_slope.size && dims_slope.size == 1) {
+    odla_value_shape slope_shape = dims_input;
+    for (int i = 0; i < dims_input.size; i++) {
+      if (dims_input.dims[i] == dims_slope.dims[0]) {
+        slope_shape.dims[i] = dims_slope.dims[0];
+      }
+      slope_shape.dims[i] = 1;
+    }
+    new_slope = CreateValue(slope->mem, slope_shape, value_id);
+  }
+
+  auto neg_slop = unary_eltwise_op(dnnl::algorithm::eltwise_linear, new_slope,
```

Comment:
It looks like "unary_eltwise_op" should support broadcasting rule, doesn't it?
PRelu itself is insensitive to NHWC/NCHW.

Reply:
yes,  if all devices have a limit that input and slope have same dims, we even can  implement it in halo

Reply:
hi hongyao, as matter of fact,  we should both check lhs and rhs value when call `binary_eltwise()` so I fix it once for all :) see #116